Backpack/src/main/java/net/skyscanner/backpack/calendar2/data/CalendarDispatchers.kt

```diff
@@ -0,0 +1,28 @@
+package net.skyscanner.backpack.calendar2.data
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import net.skyscanner.backpack.util.InternalBackpackApi
+
+@InternalBackpackApi
+@VisibleForTesting
+object CalendarDispatchers {
```

Comment:
That's the approach Kotlin Coroutines use – e.g. usually you refer to Dispatchers.Main, but you can do Dispatchers.setMain() for testing purposes. I consider this as a reasonable solution if we have such holders related to each component (e.g. avoiding god-holder patters).

Injecting something into view may be more difficult, besides it's much easier to make a mistake when you can View constructor to call internal or testing constructors – IDE will show all of the constructors. You need to type CalendarDispatchers explicitly to find this api :)

Reply:
What's the reason for not using the built in `Dispatchers.setMain` approach here, rather than creating essentially another implementation of it here? it looks like `Dispatchers.setMain` is also only available in the testing package, which would avoid somebody using it to update the application code to the main thread and we wouldn't have to have the `InternalBackpackApi` annotation for it

Reply:
I would do that, but it only allows to inject the main dispatcher :(

Reply:
In that case I'm back to my original point that it feels like we're doing something we shouldn't do - this is just for the screenshot tests, right? I prefer to keep UI tests as close to the "real" implementation as possible, so forcing everything onto the main thread might be the wrong approach

Reply:
Injecting dispatchers is a common thing in coroutines. That's why there's a way to override main dispatcher provided (along with TestCoroutineDispatcher we use here).

https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/
https://medium.com/swlh/unit-testing-with-kotlin-coroutines-the-android-way-19289838d257